Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/tests-dashboard/test-create-dialog.component.ts

```diff
@@ -16,32 +16,37 @@ export class TestCreateDialogComponent {
     test: Test;
     testNameReference: string;
     isWhiteSpaceError: boolean;
+    isButtonClicked: boolean;
+
     constructor(public dialogRef: MdDialogRef<TestCreateDialogComponent>, private testService: TestService, private snackbar: MdSnackBar, public route: Router) {
         this.test = new Test();
+        this.isButtonClicked = false;
     }
     /**
      * this method is used to add a new test
      * @param testNameRef is name of the test
      */
     AddTest(testNameRef: string) {
+        this.isButtonClicked = true;
         this.test.testName = testNameRef;
         testNameRef = testNameRef.trim();
         if (testNameRef) {
             this.testService.IsTestNameUnique(testNameRef, this.test.id).subscribe((isTestNameUnique) => {
                 if (isTestNameUnique) {
                     this.testService.addTests(this.test).subscribe((responses) => {
+                        this.isButtonClicked = false;
                         this.dialogRef.close(responses);
                         this.route.navigate(['tests/' + responses.id + '/sections']);
                     },
                         errorhandling => {
-                            this.openSnackbar('Something went wrong');
+                            this.openSnackbar('Something went wrong!');
```

Comment:
I think for error handling you will require making button clicked false. so that it can be clocked again